src/content/app/entity-viewer/variant-view/variant-image/reference-sequence-allele/ReferenceSequenceAllele.tsx

```diff
@@ -0,0 +1,146 @@
+
+import SequenceLetterBlock from '../sequence-letter-block/SequenceLetterBlock';
+
+import styles from './ReferenceSequenceAllele.module.css';
+
+type Props = {
+  sequence: string;
+  regionSliceStart: number;
+  variantStart: number; // accounts for anchor base in appropriate variant types
+  variantLength: number; // accounts for anchor base in appropriate variant types
+  variantType: string;
+  mostSevereConsequence: string;
+  hasAnchorBase: boolean;
+  isSelectedAllele: boolean;
+  onClick: () => void;
+};
+
+const ReferenceSequenceAllele = (props: Props) => {
+  const {
+    sequence,
+    regionSliceStart,
+    variantStart,
+    variantLength,
+    variantType,
+    mostSevereConsequence,
+    isSelectedAllele,
+    onClick
+  } = props;
+
+  if (variantType === 'insertion') {
+    return null;
+  }
+
+  const shouldDisplayGap = variantLength > MAX_REFERENCE_ALLELE_DISPLAY_LENGTH;
+
+  const letterBlockClasses = classNames(styles.letter);
+
+  const sequenceSliceStart = variantStart - regionSliceStart;
+  const letters = sequence
+    .slice(sequenceSliceStart, sequenceSliceStart + variantLength)
+    .split('');
+
+  const defaultLetterColour = 'var(--color-grey)'; // this is a fallback colour that should never be displayed if everything is working correctly
+  const letterBlockColour = isSelectedAllele
+    ? getVariantGroupCSSColour(mostSevereConsequence) ?? defaultLetterColour
+    : 'var(--color-blue)';
+  const letterColour = isSelectedAllele
+    ? 'var(--color-black)'
+    : 'var(--color-white)';
+  const sequenceLetterStyle: Record<string, string> = {
+    '--sequence-block-color': letterBlockColour,
+    '--sequence-block-letter-color': letterColour
+  };
+
+  if (!shouldDisplayGap) {
+    return (
+      <button onClick={onClick} disabled={isSelectedAllele}>
+        {letters.map((letter, index) => (
+          <SequenceLetterBlock
+            key={index}
+            letter={letter}
+            className={letterBlockClasses}
+            style={sequenceLetterStyle}
+          />
+        ))}
+      </button>
+    );
+  } else {
+    // show six nucleotides on both sides
+    const lettersLeft = letters
+      .slice(0, 6)
+      .map((letter, index) => (
+        <SequenceLetterBlock
+          key={index}
+          letter={letter}
+          className={letterBlockClasses}
+          style={sequenceLetterStyle}
+        />
+      ));
+    const lettersRight = letters
+      .slice(letters.length - 6)
+      .map((letter, index) => (
+        <SequenceLetterBlock
+          key={index}
+          letter={letter}
+          className={letterBlockClasses}
+          style={sequenceLetterStyle}
+        />
+      ));
+    const twoDotBlocks = [...Array(2)].map((_, index) => (
+      <SequenceLetterBlock
+        key={index}
+        letter={'.'}
+        className={letterBlockClasses}
+      />
+    ));
+    const missingSequenceLength = sequence.length - 20;
```

Comment:
Add a comment to describe why sequence length is subtracted by 20?

Reply:
✅